File: ordenamientos de java/TODO.java
import java.io.*;
import java.util.*;

public class TODO {
    public static void main(String[] args) throws IOException {
        String[] files = {
            "File_100.txt", "File_500.txt", "File_1000.txt", "File_2000.txt",
            "File_3000.txt", "File_4000.txt", "File_5000.txt", "File_6000.txt",
            "File_7000.txt", "File_8000.txt", "File_10000.txt", "File_20000.txt",
            "File_30000.txt", "File_40000.txt", "File_50000.txt", "File_60000.txt",
            "File_70000.txt", "File_80000.txt", "File_100000.txt"
        };

        for (String file : files) {
            System.out.println("\n--- Resultados para " + file + " ---");
            int[] arregloOriginal = leerArchivo(file);

            ejecutarAlgoritmo(arregloOriginal.clone(), "Bubble Sort", TODO::bubbleSort);
            ejecutarAlgoritmo(arregloOriginal.clone(), "Selection Sort", TODO::selectionSort);
            ejecutarAlgoritmo(arregloOriginal.clone(), "Insertion Sort", TODO::insertionSort);
            ejecutarAlgoritmo(arregloOriginal.clone(), "Merge Sort", arr -> mergeSort(arr, 0, arr.length - 1));
            ejecutarAlgoritmo(arregloOriginal.clone(), "Quick Sort", arr -> quickSort(arr, 0, arr.length - 1));
            ejecutarAlgoritmo(arregloOriginal.clone(), "Heap Sort", TODO::heapSort);
            ejecutarAlgoritmo(arregloOriginal.clone(), "Counting Sort", TODO::countingSort);

            System.out.println("Finalizado el procesamiento para " + file + "\n");
        }
    }

    public static void ejecutarAlgoritmo(int[] arreglo, String nombre, SortFunction algoritmo) {
        System.out.println("Ejecutando " + nombre + "...");
        long inicio = System.currentTimeMillis();
        algoritmo.sort(arreglo);
        long fin = System.currentTimeMillis();
        System.out.println(nombre + " completado en " + (fin - inicio) + " ms");
    }

    public static int[] leerArchivo(String path) throws IOException {
        BufferedReader br = new BufferedReader(new FileReader(path));
        String[] datos = br.readLine().split(" ");
        br.close();
        return Arrays.stream(datos).mapToInt(Integer::parseInt).toArray();
    }

    @FunctionalInterface
    public interface SortFunction {
        void sort(int[] arr);
    }

    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static void selectionSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIdx]) {
                    minIdx = j;
                }
            }
            int temp = arr[minIdx];
            arr[minIdx] = arr[i];
            arr[i] = temp;
        }
    }

    public static void insertionSort(int[] arr) {
        int n = arr.length;
        for (int i = 1; i < n; i++) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = key;
        }
    }

    public static void mergeSort(int[] arr, int left, int right) {
        if (left < right) {
            int mid = (left + right) / 2;
            mergeSort(arr, left, mid);
            mergeSort(arr, mid + 1, right);
            merge(arr, left, mid, right);
        }
    }

    public static void merge(int[] arr, int left, int mid, int right) {
        int n1 = mid - left + 1;
        int n2 = right - mid;
        int[] L = new int[n1];
        int[] R = new int[n2];

        System.arraycopy(arr, left, L, 0, n1);
        System.arraycopy(arr, mid + 1, R, 0, n2);

        int i = 0, j = 0, k = left;
        while (i < n1 && j < n2) {
            if (L[i] <= R[j]) {
                arr[k++] = L[i++];
            } else {
                arr[k++] = R[j++];
            }
        }

        while (i < n1) {
            arr[k++] = L[i++];
        }

        while (j < n2) {
            arr[k++] = R[j++];
        }
    }

    public static void quickSort(int[] arr, int low, int high) {
        if (low < high) {
            int pi = partition(arr, low, high);
            quickSort(arr, low, pi - 1);
            quickSort(arr, pi + 1, high);
        }
    }

    public static int partition(int[] arr, int low, int high) {
        int pivot = arr[high];
        int i = (low - 1);
        for (int j = low; j < high; j++) {
            if (arr[j] <= pivot) {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
        int temp = arr[i + 1];
        arr[i + 1] = arr[high];
        arr[high] = temp;
        return i + 1;
    }

    public static void heapSort(int[] arr) {
        int n = arr.length;
        for (int i = n / 2 - 1; i >= 0; i--) {
            heapify(arr, n, i);
        }
        for (int i = n - 1; i > 0; i--) {
            int temp = arr[0];
            arr[0] = arr[i];
            arr[i] = temp;
            heapify(arr, i, 0);
        }
    }

    public static void heapify(int[] arr, int n, int i) {
        int largest = i;
        int left = 2 * i + 1;
        int right = 2 * i + 2;
        if (left < n && arr[left] > largest) {
            largest = left;
        }
        if (right < n && arr[right] > arr[largest]) {
            largest = right;
        }
        if (largest != i) {
            int temp = arr[i];
            arr[i] = arr[largest];
            arr[largest] = temp;
            heapify(arr, n, largest);
        }
    }

    public static void countingSort(int[] arr) {
        int maxVal = Arrays.stream(arr).max().getAsInt();
        int[] count = new int[maxVal + 1];

        for (int num : arr) {
            count[num]++;
        }

        int i = 0;
        for (int num = 0; num <= maxVal; num++) {
            while (count[num] > 0) {
                arr[i++] = num;
                count[num]--;
            }
        }
    }
}

File: ordenamietos de python/TODO.py
import time
import os
from functools import partial

def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

def insertion_sort(arr):
    n = len(arr)
    for i in range(1, n):
        key = arr[i]
        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key

def merge_sort(arr):
    if len(arr) > 1:
        mid = len(arr) // 2
        L = arr[:mid]
        R = arr[mid:]

        merge_sort(L)
        merge_sort(R)

        i = j = k = 0
        while i < len(L) and j < len(R):
            if L[i] < R[j]:
                arr[k] = L[i]
                i += 1
            else:
                arr[k] = R[j]
                j += 1
            k += 1

        while i < len(L):
            arr[k] = L[i]
            i += 1
            k += 1

        while j < len(R):
            arr[k] = R[j]
            j += 1
            k += 1

def quick_sort(arr, low, high):
    if low < high:
        pi = partition(arr, low, high)
        quick_sort(arr, low, pi - 1)
        quick_sort(arr, pi + 1, high)

def partition(arr, low, high):
    pivot = arr[high]
    i = low - 1
    for j in range(low, high):
        if arr[j] <= pivot:
            i += 1
            arr[i], arr[j] = arr[j], arr[i]
    arr[i + 1], arr[high] = arr[high], arr[i + 1]
    return i + 1

def heap_sort(arr):
    n = len(arr)
    for i in range(n // 2 - 1, -1, -1):
        heapify(arr, n, i)
    for i in range(n - 1, 0, -1):
        arr[i], arr[0] = arr[0], arr[i]
        heapify(arr, i, 0)

def heapify(arr, n, i):
    largest = i
    l = 2 * i + 1
    r = 2 * i + 2
    if l < n and arr[l] > arr[largest]:
        largest = l
    if r < n and arr[r] > arr[largest]:
        largest = r
    if largest != i:
        arr[i], arr[largest] = arr[largest], arr[i]
        heapify(arr, n, largest)

def counting_sort(arr):
    max_val = max(arr)
    count = [0] * (max_val + 1)
    for num in arr:
        count[num] += 1
    i = 0
    for num in range(max_val + 1):
        while count[num] > 0:
            arr[i] = num
            i += 1
            count[num] -= 1

def leer_archivo(path):
    with open(path, 'r') as archivo:
        datos = archivo.read()
    return list(map(int, datos.split()))

files = [
    "File_100.txt", "File_500.txt", "File_1000.txt", "File_2000.txt",
    "File_3000.txt", "File_4000.txt", "File_5000.txt", "File_6000.txt",
    "File_7000.txt", "File_8000.txt", "File_10000.txt", "File_20000.txt",
    "File_30000.txt", "File_40000.txt", "File_50000.txt", "File_60000.txt",
    "File_70000.txt", "File_80000.txt", "File_100000.txt"
]

algoritmos = [
    (bubble_sort, "Bubble Sort"),
    (selection_sort, "Selection Sort"),
    (insertion_sort, "Insertion Sort"),
    (merge_sort, "Merge Sort"),
    (partial(quick_sort, low=0, high=None), "Quick Sort"),
    (heap_sort, "Heap Sort"),
    (counting_sort, "Counting Sort")
]

for file in files:
    print(f"\n--- Resultados para {file} ---")
    arreglo_original = leer_archivo(file)
    
    for alg, nombre in algoritmos:
        arreglo = arreglo_original.copy()
        print(f"Ejecutando {nombre}...")

        inicio = time.time()

        if nombre == "Quick Sort":
            alg.func(arreglo, 0, len(arreglo) - 1)
        else:
            alg(arreglo)

        fin = time.time()
        print(f"{nombre} completado en {fin - inicio:.6f} segundos")

    print(f"Finalizado el procesamiento para {file}\n")
